docs/conditions.md

```diff
@@ -0,0 +1,52 @@
+# Conditions
+
+This document defines `Conditions` and their capabilities.
+
+*NOTE*: This feature is currently a WIP.
```

Comment:
maybe link here to the related issue(s)?

Reply:
done